substrate/client/network/transactions_v2/transactions_v2_proposal.md

```diff
@@ -0,0 +1,166 @@
+2. **Introduce a New Transaction Protocol:** A more comprehensive solution may involve the development of a new transaction protocol to address the identified issues. This was discussed and proposed many times e.g. in [#6433](https://github.com/paritytech/polkadot-sdk/issues/6433).
+
+### Prior Work
+In the past an introduction of reconciliation set into Polkadot transaction protocol was mentioned many times. This approach was initially evaluated in *Erlay*.
+
+*Erlay* is a proposed enhancement to the Bitcoin protocol designed to optimize the bandwidth used during the transaction broadcast process. The protocol introduces a method where transaction hashes are initially shared with a limited set of peers, a phase known as *low fanout*. Following this, nodes request the full transaction data using a `GETDATA` message. Crucially, *Erlay* periodically triggers a set reconciliation process to synchronize transaction sets between peer pairs.
+
+While *Erlay* significantly reduces bandwidth consumption, it introduces an increase in transactions propagation latency. This trade-off needs careful consideration. Teams in Polkadot ecosystem may have different requirements and some flexibility is required in transaction protocol.
+
+For more detailed insights, consider the following resources [[1](https://delvingbitcoin.org/t/erlay-overview-and-current-approach/1415)](Sections 1-5),[[2](https://arxiv.org/pdf/1905.10518)].
+
+### Components of new transaction protocol proposal.
+
+#### Transaction Descriptor.
+The new transaction protocol introduces a flexible transaction descriptor format, living only on the network layer, supporting multiple identification schemes (including full transaction body), enabling seamless future protocol extensions with minimal rework.
+
+Proposed descriptor format (`TxDescriptor`):
+- The descriptor begins with a format specifier, allowing various types of payload:
+  - `0`: transaction body v1: `leb128(size tx body) ++ scale-encoded tx-body` (LEB128-encoded size followed by SCALE-encoded transaction body, enabling direct transmission of the full transaction body),
+  - `1`: 32-byte hash v1 (currently used),
+  - `2`: 32-bit transaction fingerprint v1, allowing further bandwidth optimizations, and laying the ground for *PinSketch* based set-reconciliation implementation as defined in *Erlay*.
+
+Two latter payloads are referred as transaction identifiers.
+
+#### Transactions Identifiers Flooding (`TxIF`).
+Transaction identifiers are gossiped to all connected peers (except _LightNodes_) allowing many transaction descriptors in a single networking notification. This reduces the required bandwidth and the depth of queues in networking module implementation. Identifiers are transmitted after a transaction is submitted to the local pool and validated as ready.
```

Comment:
Offhand this sounds good! We should probably double check with smoldot, since Pierre introduced the https://github.com/polkadot-fellows/RFCs/blob/main/text/0056-one-transaction-per-notification.md to be able to decode transactions in smoldot